src/foam/nanos/actioncommand/ActionCommandDAO.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.actioncommand;
+
+import foam.core.ClassInfo;
+import foam.core.FObject;
+import foam.core.X;
+import foam.nanos.actioncommand.ActionCommand;
+import java.lang.IllegalAccessException;
+import java.lang.NoSuchMethodException;
+import java.lang.SecurityException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ActionCommandDAO
+  extends foam.dao.ProxyDAO
+{
+  public Object cmd(Object obj) {
+    if ( obj instanceof ActionCommand ) {
+      FObject actionObj = ((ActionCommand)obj).getObject();
+      try {
+        // Calling method
+        Method action = actionObj.getClass().getDeclaredMethod(((ActionCommand)obj).getActionName());
+        action.invoke(actionObj);
+      } catch ( SecurityException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+
+      }
+    }
+    return null;
+  }
+
+  public Object cmd_(X x, Object obj) {
+    this.cmd(obj);
+    return null;
+  }
+}
```

Comment:
Since there would always be a 1:1 mapping between the services and the DAO's, it would be easier to just have the DAO be the service, via the cmd() method. Then the controller/JS-Action is assured to already have the DAO so that it can execute the remote Java actions. Also, different DAO's could be on different servers and the normal DAO routing/client-server support would handle this.